Kickstarter-iOS/Views/Controllers/ChangePasswordViewController.swift

```diff
@@ -71,13 +107,92 @@ final class ChangePasswordViewController: UIViewController {
   override func bindViewModel() {
     super.bindViewModel()
 
+    self.currentPassword.rac.text = self.viewModel.outputs.currentPasswordPrefillValue
+    self.onePasswordButton.rac.hidden = self.viewModel.outputs.onePasswordButtonIsHidden
+    self.validationErrorMessageLabel.rac.hidden = self.viewModel.outputs.validationErrorLabelIsHidden
+    self.validationErrorMessageLabel.rac.text = self.viewModel.outputs.validationErrorLabelMessage
+
+    self.viewModel.outputs.activityIndicatorShouldShow
+      .observeForUI()
+      .observeValues { shouldShow in
+        if shouldShow {
+          self.saveButtonView.startAnimating()
+        } else {
+          self.saveButtonView.stopAnimating()
+        }
+    }
+
+    self.viewModel.outputs.saveButtonIsEnabled
+      .observeForUI()
+      .observeValues { [weak self] (isEnabled) in
+        self?.saveButtonView.setIsEnabled(isEnabled: isEnabled)
+    }
+
+    self.viewModel.outputs.currentPasswordBecomeFirstResponder
+      .observeForControllerAction()
+      .observeValues { [weak self] in
+        self?.currentPassword.becomeFirstResponder()
+    }
+
+    self.viewModel.outputs.newPasswordBecomeFirstResponder
+      .observeForControllerAction()
+      .observeValues { [weak self] in
+        self?.newPassword.becomeFirstResponder()
+    }
+
+    self.viewModel.outputs.confirmNewPasswordBecomeFirstResponder
+      .observeForControllerAction()
+      .observeValues { [weak self] in
+        self?.confirmNewPassword.becomeFirstResponder()
+    }
+
+    self.viewModel.outputs.dismissKeyboard
+      .observeForControllerAction()
+      .observeValues { [weak self] in
+        self?.confirmNewPassword.resignFirstResponder()
+    }
+
+    self.viewModel.outputs.onePasswordFindPasswordForURLString
+      .observeValues { [weak self] urlString in
+        self?.onePasswordFindPassword(forURLString: urlString)
+    }
+
+    self.viewModel.outputs.changePasswordFailure
+      .observeForControllerAction()
+      .observeValues { [weak self] errorMessage in
+        self?.messageBannerView.showBanner(with: .error, message: errorMessage)
+    }
+
+    self.viewModel.outputs.changePasswordSuccess
+      .observeForControllerAction()
+      .observeValues { [weak self] params in
+        self?.logoutAndDismiss(params: params)
+    }
+
     Keyboard.change
       .observeForUI()
       .observeValues { [weak self] change in
         self?.handleKeyboardVisibilityDidChange(change)
     }
   }
 
+  // MARK: Private Functions
+  private func logoutAndDismiss(params: DiscoveryParams?) {
+    AppEnvironment.logout()
+
```

Comment:
Should we add `PushNotificationDialog.resetAllContexts()` here, since the user logged out? 🤔